plugins/trace_api_plugin/trace_api_plugin.cpp

```diff
@@ -296,6 +329,59 @@ struct trace_api_rpc_plugin_impl : public std::enable_shared_from_this<trace_api
    std::shared_ptr<request_handler_t> req_handler;
 };
 
+namespace trace_apis {
+
+   auto make_resolver(const controller& control, abi_serializer::yield_function_t yield) {
+      return chain_apis::resolver_factory::make(control, std::move( yield ));
+   }
+
+   read_only::get_transaction_result read_only::get_transaction(const read_only::get_transaction_params &p) const {
+      read_only::get_transaction_result result{};
+      transaction_id_type input_id;
+      auto input_id_length = p.id.size();
+      try {
+         FC_ASSERT( input_id_length <= 64, "hex string is too long to represent an actual transaction id" );
+         FC_ASSERT( input_id_length >= 8,  "hex string representing transaction id should be at least 8 characters long to avoid excessive collisions" );
+         input_id = transaction_id_type(p.id);
+      } EOS_RETHROW_EXCEPTIONS(transaction_id_type_exception, "Invalid transaction ID: ${transaction_id}", ("transaction_id", p.id))
+
+      fc::microseconds max_response_time = app().get_plugin<http_plugin>().get_max_response_time();
+      const auto deadline = trace_api_rpc->calc_deadline( max_response_time );
+
+      // search for the number of the block that contains the transaction
+      get_block_n blk_num;
+      blk_num = trace_api_rpc->common->store->get_trx_block_number(input_id, [deadline]() { FC_CHECK_DEADLINE(deadline); });
+
+      // extract the trx trace from the block trace
+      if (blk_num.has_value()){
+         auto resp = trace_api_rpc->req_handler->get_block_trace(blk_num.value(), [deadline]() { FC_CHECK_DEADLINE(deadline); });
+         if (resp.is_null()) {
+            EOS_THROW(tx_not_found, "Transaction ${id} not found in trace log", ("id",p.id));
+         } else {
+            auto& b_mvo = resp.get_object();
+            if (b_mvo.contains("transactions")) {
+               auto& transactions = b_mvo["transactions"];
+               for (uint32_t i = 0; i < transactions.size(); ++i) {
+                  if (transactions[i].is_null()) continue;
+                  auto& t_mvo = transactions[i].get_object();
+                  if (t_mvo.contains("id")) {
+                     string t_id = t_mvo["id"].as_string();
```

Comment:
Should be able to use `get_string()` instead of `as_string()` and not have to make a copy of the string.

Reply:
Fixed.